validation/connection_fields.xsd

```diff
@@ -25,7 +25,13 @@
     <xs:attribute name="default-value" type="xs:string"/>
     <xs:attribute name="editable" type="xs:boolean"/>
     <xs:attribute name="secure" type="xs:boolean"/>
-    <xs:attribute name="name" type="xs:string" use="required"/>
+    <xs:attribute name="name" use="required">
+     <xs:simpleType>
+        <xs:restriction base="xs:string">
+          <xs:pattern value="[a-zA-Z_]([a-zA-Z0-9_.-])*"/>
```

Comment:
Updated the regex to not require escaping. Looks like we would not need to escape - if it is the last character for the pattern.